### ISSUE TITLE
sd-boot mesure efi kernels without sd-stub

### ISSUE THREAD

**Issue by glance-** [systemd/systemd#26986]
### Component

systemd-boot

### Is your feature request related to a problem? Please describe

I'm using plain linux kernels built with efi stub, compiled in initramfs and cmdline, and those are signed with our secure boot keys.

This is to be able to keep compatibility with our old still bios booted hardware, which makes them capable of booting the exact same kernel, and which is one of the reasons we can't go sd-stub.

The downside of this is that systemd currently relies on sd-stub to measure the kernel into the tpm, and when booting a non sd-stub kernel, only the command line in the .conf file gets measured by systemd-boot.

### Describe the solution you'd like

I'd like to see systemd-boot measure the efi binary it loads into the tpm. Either it detects that it's not a UKI and mesures the efi binary into PCR 11 like sd-stub does, or it always mesures the binary it loads into some other PCR.

### The systemd version you checked that didn't have the feature you are asking for

253

**Comment by bluca** [comment]
Try using u-boot as your first stage. You can build it to boot on legacy bios and provide a full UEFI environment to the later stages.

**Comment by polarina** [comment]
I was under the impression that systemd-boot would measure EFI binaries into PCR4, since that is that register's purpose, but by looking at the source-code that does not appear to be the case. I find this to be a surprising behaviour!

**Comment by bluca** [comment]
See: https://uapi-group.org/specifications/specs/linux_tpm_pcr_registry/

**Comment by poettering** [comment]
I don't think this is desirable. PCR4 measurements exist anyway by firmware. And given your kernel is a PE kernel anyway you might as well just stick sd-stub in front of it if you also want PCR 11. You don't have to populate the .cmdline or .initrd if you don't want to after all. Seems a lot more robust, then doing the same measurements at two places.

Hence, closing.
